### ISSUE TITLE
[Bug]

### ISSUE THREAD

**Issue by alessandro-bottamedi** [ammarahm-ed/react-native-mmkv-storage#113]
**Describe the bug**
Multiple crash reported on android

**To Reproduce**
i dont'know, crash reported from google play console...

**Platform Information:**
 - OS: Android
 - React Native Version: 0.63.4
 - Library Version: 0.5.9

**Log Stack**
```
java.lang.NullPointerException: 
  at com.facebook.react.bridge.Arguments.fromBundle (Arguments.java:291)
  at com.ammarahmed.mmkv.RNMMKVModule.writeToJSON (RNMMKVModule.java:122)
  at com.ammarahmed.mmkv.RNMMKVModule.migrate (RNMMKVModule.java:107)
  at com.ammarahmed.mmkv.RNMMKVModule.initialize (RNMMKVModule.java:79)
  at com.facebook.react.bridge.ModuleHolder.doInitialize (ModuleHolder.java:236)
  at com.facebook.react.bridge.ModuleHolder.markInitializable (ModuleHolder.java:100)
  at com.facebook.react.bridge.NativeModuleRegistry.notifyJSInstanceInitialized (NativeModuleRegistry.java:103)
  at com.facebook.react.bridge.CatalystInstanceImpl$4.run (CatalystInstanceImpl.java:541)
  at android.os.Handler.handleCallback (Handler.java:883)
  at android.os.Handler.dispatchMessage (Handler.java:100)
  at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage (MessageQueueThreadHandler.java:27)
  at android.os.Looper.loop (Looper.java:237)
  at com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run (MessageQueueThreadImpl.java:226)
  at java.lang.Thread.run (Thread.java:919)
```

**Comment by ammarahm-ed** [comment]
Hey @alessandro-bottamedi Did you upgrade from 0.4.x to 0.5.x in your app and is it happening on all devices or some only?

**Comment by alessandro-bottamedi** [comment]
Yes, why? Error is happening on some device only, perhaps those on which there was old data?

Is there a way to recognize old data and in this case delete them?

**Comment by ammarahm-ed** [comment]
@alessandro-bottamedi The library is trying to migrate data which is causing a crash. Are you sure it is happening on all devices that have old data or only some with old data? 

What you can do is, run the old version app after building from source of old version. do something so there is data in storage. then upgrade the same app to new version. let me know if the app crashes or not. I will try to handle the exception in the mean time.

**Comment by ammarahm-ed** [comment]
I think I have found the solution. I will fix it today.

**Comment by alessandro-bottamedi** [comment]
> @alessandro-bottamedi The library is trying to migrate data which is causing a crash. Are you sure it is happening on all devices that have old data or only some with old data? 
> 
> 
> 
> What you can do is, run the old version app after building from source of old version. do something so there is data in storage. then upgrade the same app to new version. let me know if the app crashes or not. I will try to handle the exception in the mean time.
> 
> 
> 
> 

I will try tomorrow morning with an old version, I will let you know.

**Comment by alessandro-bottamedi** [comment]
No errors in android emulator (upgrade with old data from 0.4.3 to 0.5.9), is it possible that the error occurs only with real devices? Or only with specific data types?

**Comment by ammarahm-ed** [comment]
@alessandro-bottamedi The error happens when you for example set an object to storage then call, `MMKV.setMapAsync("key",null)`. When migrating from 0.4.x to 0.5.x this is causing issue. I have fixed it on master. You can test it like this:

The error can happen with maps or arrays possibly.

On old version set an object to storage then set its value to null. upgrade library and test.
```js
`MMKV.setMapAsync("key",{hello:'world"})`
`MMKV.setMapAsync("key",null)`
```
install new version and run app. If app loads, the issue is fixed.

Edit: Install from master branch

**Comment by alessandro-bottamedi** [comment]
Thanks, it seems that the problem is fixed, I give the new version to some testers. I will confirm in the next days if it has been solved.

**Comment by ammarahm-ed** [comment]
I will close this now. it's released in v0.6.0. reopen if problem occurs

**Comment by alessandro-bottamedi** [comment]
I confirm that the problem no longer occurs thanks
